rust/lance/src/index/vector/ivf.rs

```diff
@@ -779,12 +748,23 @@ pub async fn build_ivf_pq_index(
         } else {
             training_data
         };
+        let training_data = Arc::new(training_data);
+
+        // TODO: consolidate IVF models to `lance_index`.
+        let ivf2 = lance_index::vector::ivf::Ivf::new(
+            ivf_model.centroids.as_ref().try_into()?,
+            metric_type,
+            vec![],
+        );
         // Compute the residual vector to train Product Quantizer.
-        let part_ids = span!(Level::INFO, "compute partition")
-            .in_scope(|| ivf_model.compute_partitions(&training_data, metric_type));
+        let part_ids = ivf2
+            .compute_partitions(training_data.clone())
+            .instrument(span!(Level::INFO, "compute partition for PQ training"))
```

Comment:
Minor nit, this is correct, but could you put the `instrument` call inside of `compute_partitions`?  `instrument` needs to accompany `spawn_blocking` so having those two calls in the same scope makes it easier for future readers to understand why it is there.

Reply:
Sure thing.